<launch>
  <arg name="scan_topic" default="/base_scan"/>
  <node name="laser_scan_matcher_node" pkg="laser_scan_matcher"
    type="laser_scan_matcher_node" output="screen">
    <remap from="/scan" to="$(arg scan_topic)"/>
    <param name="use_odom" value="false"/>
    <param name="use_imu" value="false"/>
    <param name="fixed_frame" value="odom"/>
    <param name="base_frame" value="base_footprint"/>
    <param name="publish_tf" value="false"/>
    <param name="publish_pose_stamped" value="false"/>
    <param name="max_iterations" value="30"/>
    <param name="max_correspondence_dist" value="0.2"/>
    <param name="max_angular_correction_deg" value="65"/>
    <param name="max_linear_correction" value="0.5"/>
    <param name="outliers_maxPerc" value="0.98"/>
    <param name="kf_dist_linear" value="0.2"/>
    <param name="kf_dist_angular" value="0.2"/>
    <param name="orientation_neighbourhood" value="20"/>

    <param name="outliers_adaptive_order" value="0.98"/>

    <param name="do_compute_covariance" value="1"/>
    <param name="publish_pose_with_covariance_stamped" value="true"/>
  </node>
</launch>
